packages/three-vrm-materials-mtoon/src/nodes/MToonNodeMaterial.ts

```diff
@@ -0,0 +1,485 @@
+    this.rimLightingMixFactor = 1.0;
+    this.rimMultiplyTexture = null;
+    this.matcapFactor = new THREE.Color(1.0, 1.0, 1.0);
+    this.matcapTexture = null;
+    this.parametricRimColorFactor = new THREE.Color(0.0, 0.0, 0.0);
+    this.parametricRimLiftFactor = 0.0;
+    this.parametricRimFresnelPowerFactor = 5.0;
+    this.outlineWidthMode = MToonMaterialOutlineWidthMode.None;
+    this.outlineWidthMultiplyTexture = null;
+    this.outlineWidthFactor = 0.0;
+    this.outlineColorFactor = new THREE.Color(0.0, 0.0, 0.0);
+    this.outlineLightingMixFactor = 1.0;
+    this.uvAnimationScrollXSpeedFactor = 0.0;
+    this.uvAnimationScrollYSpeedFactor = 0.0;
+    this.uvAnimationRotationSpeedFactor = 0.0;
+    this.uvAnimationMaskTexture = null;
+
+    this.shadeColorNode = null;
+    this.shadingShiftNode = null;
+    this.shadingToonyNode = null;
+    this.rimLightingMixNode = null;
+    this.rimMultiplyNode = null;
+    this.matcapNode = null;
+    this.parametricRimColorNode = null;
+    this.parametricRimLiftNode = null;
+    this.parametricRimFresnelPowerNode = null;
+
+    this.uvAnimationScrollXOffset = 0.0;
+    this.uvAnimationScrollYOffset = 0.0;
+    this.uvAnimationRotationPhase = 0.0;
+
+    this.isOutline = false;
+
+    this._animatedUVNode = null;
+
+    this.setValues(parameters);
+  }
+
+  public setupLightingModel(/*builder*/): MToonLightingModel {
+    return new MToonLightingModel();
+  }
+
+  public setup(builder: Nodes.NodeBuilder): void {
+    this._animatedUVNode = new MToonAnimatedUVNode(
+      (this.uvAnimationMaskTexture && this.uvAnimationMaskTexture.isTexture === true) ?? false,
+    );
+
+    super.setup(builder);
+  }
+
+  public setupDiffuseColor(builder: Nodes.NodeBuilder): void {
+    // we must apply uv scroll to the map
+    // this.colorNode will be used in super.setupDiffuseColor() so we temporarily replace it
+    let tempColorNode: Nodes.ShaderNodeObject<Nodes.Node> | null = null;
+
+    if (this.colorNode == null) {
+      tempColorNode = refColor;
+
+      if (this.map && this.map.isTexture === true) {
+        const map = refMap.context({ getUV: () => this._animatedUVNode });
+        tempColorNode = tempColorNode.mul(map);
+      }
+
+      this.colorNode = tempColorNode;
+    }
+
+    // MToon must ignore vertex color by spec
+    // See: https://github.com/vrm-c/vrm-specification/blob/42c0a90e6b4b710352569978f14980e9fc94b25d/specification/VRMC_materials_mtoon-1.0/README.md#vertex-colors
+    if (this.vertexColors === true && builder.geometry.hasAttribute('color')) {
+      console.warn(
+        'MToonNodeMaterial: MToon ignores vertex colors. Consider using a model without vertex colors instead.',
+      );
+      this.vertexColors = false;
+    }
+
+    // the ordinary diffuseColor setup
+    super.setupDiffuseColor(builder);
+
+    // Set alpha to 1 if it is not transparent
+    if (this.transparent === false) {
+      Nodes.diffuseColor.a.assign(1.0);
+    }
```

Comment:
I need to confirm the conditions for making it opaque.

https://github.com/pixiv/three-vrm/blob/9e16c3855a5b441cca0f11480c4003de25d2943c/packages/three-vrm-materials-mtoon/src/shaders/mtoon.frag#L871-L873

https://github.com/mrdoob/three.js/blob/c8ceac2f12655156472e3f9b80a63d8baf4bfdf9/src/renderers/webgl/WebGLPrograms.js#L245

Reply:
changed that to follow the WebGLRenderer way for now.

https://github.com/pixiv/three-vrm/pull/1384/commits/afe9143d2054097c1b99aae876a77d4bbe58c38c

Reply:
NICE!